lib/proxy_api/resource.rb

```diff
@@ -63,15 +63,22 @@ def parse(response)
       false
     end
 
+    def escape_path(path)
+      parsed_path = URI.parse(path)
+      query = parsed_path.query
+      path = "#{parsed_path.path}?#{CGI.escape(query)}" unless query.nil?
+      path
+    end
+
     # Perform GET operation on the supplied path
-    def get(path = nil, payload = {})
+    def get(path = nil, additional_headers = {})
       with_logger do
         telemetry_duration_histogram(:proxy_api_duration, :ms, method: 'get') do
           # This ensures that an extra "/" is not generated
```

Comment:
This comment seems to indicate this escaping might not what we actually want to do here?